packages/core/saved-objects/core-saved-objects-migration-server-internal/src/actions/es_errors.ts

```diff
@@ -17,7 +17,8 @@ export const isWriteBlockException = (errorCause?: estypes.ErrorCause): boolean
 export const isIncompatibleMappingException = (errorCause?: estypes.ErrorCause): boolean => {
   return (
     errorCause?.type === 'strict_dynamic_mapping_exception' ||
-    errorCause?.type === 'mapper_parsing_exception'
+    errorCause?.type === 'mapper_parsing_exception' ||
+    errorCause?.type === 'document_parsing_exception'
```

Comment:
Indexing a document with mapping mismatch (between the doc's fields and the index's mapping) now return a `document_parsing_exception` exception.